src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -122,6 +128,11 @@ public void createBuildPlanForExercise(ProgrammingExercise exercise, String plan
     private void setupGitLabCIConfiguration(VcsRepositoryUri repositoryUri, ProgrammingExercise exercise, String buildPlanId) {
         final String repositoryPath = uriService.getRepositoryPathFromRepositoryUri(repositoryUri);
         ProjectApi projectApi = gitlab.getProjectApi();
+
+        if (exercise.getBuildConfig() == null || !Hibernate.isInitialized(exercise.getBuildConfig())) {
+            exercise.setBuildConfig(programmingExerciseBuildConfigRepository.getProgrammingExerciseBuildConfigElseThrow(exercise));
+        }
```

Comment:
This check is not needed since it is done in the repository method. 

Maybe you can even create a `loadAndSet` method since you use calls like here in a few other places.